### DISCUSSION TITLE
Use cargo nightly

### DISCUSSION THREAD

**Discussion by mercxry** [renovatebot/renovate#15728] [Request Help]
Hello, I'm using a feature for Cargo `workspace-inheritance` that requires the Nightly channel, is it possible to set that in the configuration somehow?

I'm using the hosted instance of renovate but if it's not possible, I'm open to self-host an instance for this purpose.

**Comment by rarkins** [comment]
In the hosted app, Renovate uses the `renovate/rust` pre-built Docker images: https://hub.docker.com/r/renovate/rust/tags

Looking at the code which invokes it, it seems like we would always use `latest` though: https://github.com/renovatebot/renovate/blob/51983718b95b9bcb8e312a1171b1d00b6d134429/lib/modules/manager/cargo/artifacts.ts#L13-L30

Although Renovate is not yet capable of using nightly, if it were to be then how could it know to? i.e. is there something in your Cargo config in the repo which the bot could use to know you need nightly and not stable/latest?

You could build your own Renovate image, the challenge is that when you build it you could add cargo nightly, but then a few days later it's out of date so you'd need to somehow keep building the images.

**Reply by mercxry** [reply]
Yes, I use a `rust-toolchain.toml` file to enforce nightly, you can find the [docs here](https://rust-lang.github.io/rustup/overrides.html#the-toolchain-file), but mine looks like this:
```toml
[toolchain]
channel = "nightly"
```

I've seen that there is an [issue open](https://github.com/renovatebot/renovate/issues/6365) about this, I could create a minimal reproduction repo for this, would that help?

**Comment by folex** [comment]
It seems that `workspace-inheritance` is stabilized in 1.64. But renovate still has Rust 1.63 in [latest docker](https://hub.docker.com/layers/renovate/rust/latest/images/sha256-d33bfd5e33a7dae848bf3246a19ec5b2677f7fc43673ce4bc94fa6de75f6bc44?context=explore). 

What's the good way to bump it to help bump it to 1.64?

**Reply by rarkins** [reply]
It is built from here: https://github.com/renovatebot/docker-renovate-full/blob/d422882fea77b2d7d1ba1f11fa0cfa974ce45cce/Dockerfile#L88-L89

It will be updated automatically by Renovate once 1.64.0 is discovered (published to Docker Hub: https://hub.docker.com/_/rust)
